src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -377,6 +391,21 @@ public void testShouldNotThrowIfReporterDoesNotExist() {
     setUpTask();
   }
 
+  @Test
+  public void testShouldVerifyChangingTopic() {
+    props.put(FLUSH_SYNCHRONOUSLY_CONFIG, "false");
+    setUpTask();
+    String changedTopic = "routed-to-another-topic";
+    SinkRecord record = record(changedTopic, false, false, 0);
+    when(context.assignment()).thenReturn(Collections.singleton(new TopicPartition("original-topic-name", 1)));
+    try {
```

Comment:
This can be simplified as 
```
ConnectException ce = assertThrows(()->task.put(Collections.singletonList(record)));
assertEquals(ce.getMessage(), String.format("Found a topic name '%s' that doesn't match assigned partitions."
          + " Connector doesn't support topic mutating SMTs", record.topic()));
```